libs/services/designer-client-services/src/lib/base/function.ts

```diff
@@ -53,4 +54,120 @@ export class BaseFunctionService implements IFunctionService {
     });
     return keysResponse?.default ?? 'NotFound';
   }
+
+  async fetchSwaggerUrl(functionAppId: string) {
+    const { baseUrl } = this.options;
+    const response = await this.options.httpClient.get<any>({
+      uri: `${baseUrl}/${functionAppId}/config/web`,
+      queryParameters: { 'api-version': this.options.apiVersion },
+    });
+    if (!response?.properties?.apiDefinition?.url) {
+      throw new Error('ApiDefinitionUrl not found');
+    }
+    return response.properties.apiDefinition.url;
+  }
+
+  private async fetchFunctionSwagger(swaggerUrl: string) {
+    const response = await this.options.httpClient.get<any>({
+      uri: swaggerUrl,
+      headers: { 'Access-Control-Allow-Origin': '*' },
+    });
+    const swaggerDoc = await SwaggerParser.parse(response);
+    return new SwaggerParser(swaggerDoc);
+  }
+
+  public async fetchFunctionAppSwagger(functionAppId: string) {
+    const apiDefinitionUrl = await this.fetchSwaggerUrl(functionAppId);
+    return this.fetchFunctionSwagger(apiDefinitionUrl);
+  }
+
+  public async getOperationSchema(functionAppId: string, operationId: string, isInput: boolean): Promise<any> {
+    const swagger = await this.fetchFunctionAppSwagger(functionAppId);
+    if (!operationId) return Promise.resolve();
+    const operation = swagger.getOperationByOperationId(operationId);
+    if (!operation) throw new Error('Operation not found');
+
+    const paths = swagger.api.paths[operation.path];
+    const rawOperation = paths[operation.method];
+    const schema = { type: 'object', properties: {} as any, required: [] as string[] };
+
+    if (isInput) {
+      const baseUrl = swagger.api.host
+        ? swagger.api.schemes?.length
+          ? `${swagger.api.schemes.at(-1)}://${swagger.api.host}`
+          : `http://${swagger.api.host}`
+        : swagger?.api?.basePath ?? 'NotFound';
```

Comment:
this is not including the basepath

Reply:
What do you mean by including the basepath

Reply:
this chunk of code is not getting this - 

schemes + api.host + api.basePath

YOu can either have baseUrl as the above or just have baseUrl as schemes + api.host and in the below comment have 
default: api.basePath ? baseurl + api.basepath + operation.path : baseurl + operation.path
See appservices.ts